LFPy/recextelectrode.py

```
@@ -188,51 +232,76 @@ def __init__(self, cell=None, sigma=0.3,
             self.sigma = sigma
             self.anisotropic = False
 
-        if type(x) in [float, int]:
-            self.x = np.array([x])
-        else:
-            self.x = np.array(x).flatten()
-        if type(y) in [float, int]:
-            self.y = np.array([y])
-        else:
-            self.y = np.array(y).flatten()
-        if type(z) in [float, int]:
-            self.z = np.array([z])
-        else:
-            self.z = np.array(z).flatten()
-        try:
-            assert((self.x.size==self.y.size) and (self.x.size==self.z.size))
-        except AssertionError:
-            raise AssertionError("The number of elements in [x, y, z] must be identical")
-
-        if N is not None:
-            if type(N) != np.array:
-                try:
-                    N = np.array(N)
-                except:
-                    print('Keyword argument N could not be converted to a '
-                          'numpy.ndarray of shape (n_contacts, 3)')
-                    print(sys.exc_info()[0])
-                    raise
-            if N.shape[-1] == 3:
-                self.N = N
+        if probe is None:
+            if x is None:
+                x = np.array([0])
+            if y is None:
+                y = np.array([0])
+            if z is None:
+                z = np.array([0])
+            if type(x) in [float, int]:
+                self.x = np.array([x])
```

I think we can raise an error if all of `probe` and `x,y,z` is `None`, or if both sets are set. Not sure why we set a the default as it were anymore.

---

ok